### PR TITLE
Qt6

### PR DESCRIPTION
**Pull request by Ri0n**

### CODE REVIEW

File: generic/extendedmenuplugin/extendedmenuplugin.cpp [hunk 1]
```diff
@@ -265,8 +265,12 @@ static const QString secondsToString(ulong seconds)
 
 static int stringToInt(const QString &str)
 {
-    int sign   = (str.at(0) == '-') ? -1 : 1;
+    int sign = (str.at(0) == '-') ? -1 : 1;
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
     int result = str.midRef(1, 2).toInt() * sign;
+#else
+    int result = str.mid(1, 2).toInt() * sign;
```
Comment (Ri0n) [inline generic/extendedmenuplugin/extendedmenuplugin.cpp:272]: it's better to go with QStringView{str}.mid

File: generic/juickplugin/juickparser.cpp [hunk 2]
```diff
@@ -74,114 +75,96 @@ JuickParser::JuickParser(QDomElement *elem) : elem_(elem)
...
-#else
-                       d->singleMsgRx.cap(2).trimmed().split(" ", QString::SkipEmptyParts), d->singleMsgRx.cap(3),
-#endif
-                       d->singleMsgRx.cap(6), d->singleMsgRx.cap(5));
+        match     = d->singleMsgRx.match(msg);
+        JuickMessage m(match.captured(1), match.captured(4), match.captured(2).trimmed().split(" ", Qt::SkipEmptyParts),
+                       match.captured(3), match.captured(6), match.captured(5));
         messages_.append(m);
-    } else if (d->juboRx.indexIn(msg) != -1) {
+    } else if (d->juboRx.match(msg).hasMatch()) {
```
Comment (Ri0n) [inline generic/juickplugin/juickparser.cpp:156]: better try something like 
```
if (match = d->juboRx.match(msg), match.hasMatch()) {
```
or 
```
if ((match = d->juboRx.match(msg)).hasMatch()) {
```
so no need to match again

File: generic/extendedmenuplugin/extendedmenuplugin.cpp (hunk 1 above)
Comment (Vitozz) [reply generic/extendedmenuplugin/extendedmenuplugin.cpp]: resolved

File: generic/juickplugin/juickparser.cpp (hunk 2 above)
Comment (Vitozz) [reply generic/juickplugin/juickparser.cpp]: resolved

**Pull request merged**